src/code_gen.rs

```diff
@@ -235,64 +232,40 @@ impl<'a> TreeFold<'a> for CodeGen<'a> {
     fn visit_action(&mut self, action: &'a Action) {
         match action {
             Action::GetProperty(id, p) => {
-                if id.id_name == "target" {
-                    if p.id_name == "height" {
-                        let cpp_var_id = "get_tree_height_target";
-
-                        // NOTE: I chose to use rust format! macro instead of handlebars templating
-                        // for the ease of testing and faster iteration.
-                        let block = format!(
-                            "std::string {cpp_var_id} = std::to_string({func_name}({args}));",
-                            cpp_var_id = cpp_var_id,
-                            func_name = "get_tree_height",
-                            args = "target"
-                        );
-
-                        self.blocks.push(block);
-
-                        self.result = CppResult::Return {
-                            typ: CppType::Int,
-                            id: String::from(cpp_var_id),
-                        };
-                    } else {
-                        panic!("{} graph property not supported", p.id_name)
-                    }
-                } else {
-                    let attribute = &self.config.attributes_to_property_parts[p.id_name];
-
-                    let cpp_var_id: String =
-                        String::from(id.id_name) + "_" + &attribute.parts.join("_");
-
-                    let mut parts = String::from("{");
-                    parts.push_str(
-                        &attribute
-                            .parts
-                            .iter()
-                            .map(|s| String::from("\"") + s + "\"")
-                            .collect::<Vec<String>>()
-                            .join(", "),
-                    );
-                    parts.push_str("}");
-
-                    let block = format!(
-                        "node_ptr = get_node_with_id(target, mapping->at(\"{node_id}\"));
+                let attribute = &self.config.attributes_to_property_parts[p.id_name];
```

Comment:
A few comments explaining what is going on here would be useful, both at this location and the locations below.

Reply:
It's just stored in memory, not sent to backend.

Reply:
Implemented moving average, and added comments here.